hibernate-core/src/main/java/org/hibernate/engine/spi/LoadQueryInfluencers.java

```diff
@@ -183,18 +187,24 @@ public void disableFetchProfile(String name) throws UnknownProfileException {
 	}
 
 	public EntityGraph getFetchGraph() {
-		return fetchGraph;
+		return entityGraphSemantic == EntityGraphSemantic.FETCH
+				? entityGraph
+				: null;
 	}
 
 	public void setFetchGraph(final EntityGraph fetchGraph) {
-		this.fetchGraph = fetchGraph;
+		this.entityGraphSemantic = EntityGraphSemantic.FETCH;
```

Comment:
To an outsider setFetchGraph and setLoadGraph may sound independent but we know they aren't now. The effect is also visible via the getters. "Why did A disappear if I set B". Would it not be more "honest" and clear to just accept this way all the say and have just a "set[Entity]Graph" that takes two parameters - EntityGraph and EntityGraphSemantic and a getter for each - get[Entity]Graph and get[Entity]GraphSemantic?

Alternatively maybe differently named setters could work. Something like "set**As**FetchGraph" and "set**As**LoadGraph" but getters don't benefit from that.